### ISSUE TITLE
Date generators should allow for specific dates to be selected

### ISSUE THREAD

**Issue by LeoColman** [kotest/kotest#1354]
Today, when using an Arb such as the arb for local date, the only configurable parameter is the year:

```
Arb.localDate(minYear = 1970, maxYear = 2020)
```

We should be allowed to generate a date with a specific date as start/end:

```
Arb.localDate(minDate = LocalDate.of(1970, 01, 01), maxDate = LocalDate.of(2020, 04, 05)
```

**Comment by sksamuel** [comment]
Will you add a PR ?

**Comment by LeoColman** [comment]
This was merged already at #1404
